src/lib.rs

```diff
@@ -54,347 +58,456 @@ impl Default for Duration {
+            ListParam::IexPercent => String::from("iexpercent"),
+        }
     }
+}
 
-    pub fn logo(&self, symbol: &str) -> Result<Logo> {
-        self.get(&format!("/stock/{}/logo", symbol))
-    }
+#[derive(PartialEq, Eq)]
+pub enum ChartParam {
+    /// boolean. If true, 1d chart will reset at midnight instead of the default behavior of 9:30am ET.
+    Reset(bool),
+    /// boolean. If true, runs a polyline simplification using the Douglas-Peucker algorithm. This is useful if plotting sparkline charts.
+    Simplify(bool),
+    /// number. If passed, chart data will return every Nth element as defined by `Interval`.
+    Interval(usize),
+    /// boolean. If true, changeOverTime and marketChangeOverTime will be relative to previous day close instead of the first value.
+    ChangeFromClose(bool),
+    /// number. If passed, chart data will return the last N elements.
+    Last(usize),
+}
 
-    pub fn news(&self, symbol: &str) -> Result<Vec<News>> {
-        // TODO: this also takes a count argument, implement it.
-        self.get(&format!("/stock/{}/news", symbol))
+impl ToString for ChartParam {
+    fn to_string(&self) -> String {
+        match self {
+            ChartParam::Reset(res) => format!("chartReset={}", res),
+            ChartParam::Simplify(res) => format!("chartSimplify={}", res),
+            ChartParam::Interval(res) => format!("chartInterval={}", res),
+            ChartParam::ChangeFromClose(res) => format!("changeFromClose={}", res),
+            ChartParam::Last(res) => format!("chartLast={}", res),
+        }
     }
+}
 
-    pub fn ohlc(&self, symbol: &str) -> Result<OHLC> {
-        self.get(&format!("/stock/{}/ohlc", symbol))
+/// parse_params is a function to add a query to a base url.
+///
+/// # Examples
+///
+/// ```
+/// use iex::{ChartParam, parse_params};
+///
+/// assert_eq!("?chartReset=true", parse_params(&Some(vec!(ChartParam::Reset(true)))));
+/// ```
+// REVIEW|QUESTION(): Not sure why &Option is required rather than Option.
+pub fn parse_params(params: &Option<Vec<ChartParam>>) -> String {
+    if params.is_none() {
+        return String::from("");
+    }
+    let mut result = String::from("?");
+
+    for (i, param) in params.as_ref().unwrap().iter().enumerate() {
+        if i != 0 {
+            result += "&";
+        }
+        result += &param.to_string();
     }
+    result
+}
 
-    pub fn peers(&self, symbol: &str) -> Result<Vec<String>> {
-        self.get(&format!("/stock/{}/peers", symbol))
-    }
+#[cfg(test)]
+mod tests {
+    use super::*;
 
-    pub fn previous(&self, symbol: &str) -> Result<Previous> {
-        // TODO: It's possible to pass in "market" as an argument here
-        // and get one entry for each symbol. We need to handle that
-        // scenario.
-        self.get(&format!("/stock/{}/previous", symbol))
-    }
+    // NOTE(test): Unfortunately rust has no implementation of sub-testing, or at least not aware of as of yet.
```

Comment:
Not quite sure what this comment means...

Reply:
Just informational. Originally I thought it would be best to take all of the tests that I have implemented in the PR and have them under one function. However, when I first ran that specific function, it did not tell me which assert failed, which is why I broke that one function into 20+